### DISCUSSION TITLE
How to get webots eixt message?

### DISCUSSION THREAD

**Discussion by sf-ccc** [cyberbotics/webots#6450] [General]
Hi,
i have a question that webots always show a warning about my controller is still running.I wonder how to get webots exit message to destroy my controller resources.I found supervisor has such a event which named EVENT_QUIT,how could i get this? Maybe  api or other method?

**Comment by omichel** [comment]
When Webots quits or need to terminate a controller (due to a close, reload or revert operation), the [wb_robot_step](https://cyberbotics.com/doc/reference/robot#wb_robot_step) function returns `-1`, which means that your controller should quit the control loop, proceed to destroy its resources and quit as well.
